pkg/kubelet/kuberuntime/labels.go

```diff
@@ -0,0 +1,264 @@
+		Annotations: annotations,
+	}
+
+	if podSandboxInfo.PodDeletionGracePeriod, err = getInt64PointerFromLabel(annotations, kubernetesPodDeletionGracePeriodLabel); err != nil {
+		glog.Errorf("Unable to get %q from annotations %q: %v", kubernetesPodDeletionGracePeriodLabel, annotations, err)
+	}
+	if podSandboxInfo.PodTerminationGracePeriod, err = getInt64PointerFromLabel(annotations, kubernetesPodTerminationGracePeriodLabel); err != nil {
+		glog.Errorf("Unable to get %q from annotations %q: %v", kubernetesPodTerminationGracePeriodLabel, annotations, err)
+	}
+
+	// Remove grace period
+	delete(podSandboxInfo.Annotations, kubernetesPodTerminationGracePeriodLabel)
+	delete(podSandboxInfo.Annotations, kubernetesPodDeletionGracePeriodLabel)
+
+	return podSandboxInfo
+}
+
+// getContainerInfoFromLabels gets labeledContainerInfo from labels.
+func getContainerInfoFromLabels(labels map[string]string) *labeledContainerInfo {
+	var err error
+	containerInfo := &labeledContainerInfo{
+		PodName:      getStringValueFromLabel(labels, types.KubernetesPodNameLabel),
+		PodNamespace: getStringValueFromLabel(labels, types.KubernetesPodNamespaceLabel),
+		PodUID:       kubetypes.UID(getStringValueFromLabel(labels, types.KubernetesPodUIDLabel)),
+		Name:         getStringValueFromLabel(labels, types.KubernetesContainerNameLabel),
+		TerminationMessagePath: getStringValueFromLabel(labels, kubernetesContainerTerminationMessagePathLabel),
+	}
+	if containerHash, err := getUint64ValueFromLabel(labels, kubernetesContainerHashLabel); err != nil {
+		logContainerError(containerInfo, kubernetesContainerHashLabel, err)
+	} else {
+		containerInfo.Hash = containerHash
+	}
+
+	if containerInfo.RestartCount, err = getIntValueFromLabel(labels, kubernetesContainerRestartCountLabel); err != nil {
+		logContainerError(containerInfo, kubernetesContainerRestartCountLabel, err)
+	}
+	if containerInfo.PodDeletionGracePeriod, err = getInt64PointerFromLabel(labels, kubernetesPodDeletionGracePeriodLabel); err != nil {
+		logContainerError(containerInfo, kubernetesPodDeletionGracePeriodLabel, err)
+	}
+	if containerInfo.PodTerminationGracePeriod, err = getInt64PointerFromLabel(labels, kubernetesPodTerminationGracePeriodLabel); err != nil {
+		logContainerError(containerInfo, kubernetesPodTerminationGracePeriodLabel, err)
+	}
+	preStopHandler := &api.Handler{}
+	if found, err := getJSONObjectFromLabel(labels, kubernetesContainerPreStopHandlerLabel, preStopHandler); err != nil {
+		logContainerError(containerInfo, kubernetesContainerPreStopHandlerLabel, err)
+	} else if found {
+		containerInfo.PreStopHandler = preStopHandler
+	}
+	return containerInfo
+}
+
+func getStringValueFromLabel(labels map[string]string, label string) string {
+	if value, found := labels[label]; found {
+		return value
+	}
+	// Do not report error, because there should be many old containers without label now.
+	glog.V(3).Infof("Container doesn't have label %s, it may be an old or invalid container", label)
```

Comment:
The latest labels are added in 1.2 https://github.com/kubernetes/kubernetes/commit/884dc5c9e5ef7df0679e68a6a98c7ac1d8bcffa4#diff-3982f6b6e2fa8c2c6119f1750e33cc1f. The new runtime interface will be in 1.4.

What is our deprecate policy, can we assume that all labels should be there now? And return error if not.